Sources/Apollo/GraphQLResult.swift

```diff
@@ -1,15 +1,24 @@
+/// Represents source of data in GraphQLResult
+public enum GraphQLDataSource {
```

Comment:
I think `GraphQLResult.Source` (so nesting the `Source` enum in `GraphQLResult`) would work better here.